api/app/Models/PoolCandidate.php

```diff
@@ -911,4 +919,162 @@ public function setApplicationSnapshot()
+                        'step' => $stepId,
+                        'decision' => null,
+                    ];
+
+                    continue;
+                }
+
+                foreach ($educationResults as $educationResult) {
+                    $decision = $educationResult->assessment_decision;
+                    switch ($decision) {
+                        case null:
+                            $hasToAssess = true;
+                            break;
+                        case AssessmentDecision::HOLD->name:
+                            $hasOnHold = true;
+                            break;
+                        case AssessmentDecision::UNSUCCESSFUL->name:
+                            $hasFailure = true;
+                            break;
+                        default:
+                    }
+                }
+            }
+
+            if ($hasFailure) {
+                $decisions[] = [
+                    'step' => $stepId,
+                    'decision' => AssessmentDecision::UNSUCCESSFUL->name,
+                ];
+
+                continue;
+            }
+
+            if ($hasToAssess) {
+                $decisions[] = [
+                    'step' => $stepId,
+                    'decision' => null,
+                ];
+
+                continue;
+            }
+
+            // Candidate has been assessed and was not unsuccessful so continue to next step
+            $currentStep++;
+
+            if ($hasOnHold) {
+                $decisions[] = [
+                    'step' => $stepId,
+                    'decision' => AssessmentDecision::HOLD->name,
+                ];
+
+                continue;
+            }
+
+            $decisions[] = [
+                'step' => $stepId,
+                'decision' => AssessmentDecision::SUCCESSFUL->name,
+            ];
+        }
+
+        $totalSteps = $this->pool->assessmentSteps->count();
+        $finalDecision = AssessmentFinalDecision::TO_ASSESS->name;
+
+        if ($currentStep > $totalSteps) {
+            $lastStepDecision = end($decisions);
+            if ($lastStepDecision['decision'] !== AssessmentDecision::HOLD->name) {
+                $finalDecision = AssessmentFinalDecision::QUALIFIED->name;
+                $currentStep = null;
+            }
+        } else {
+            $unsuccessfulDecisions = Arr::where($decisions, function ($stepDecision) {
+                return $stepDecision['decision'] === AssessmentDecision::UNSUCCESSFUL->name;
+            });
+            if (! empty($unsuccessfulDecisions)) {
+                $finalDecision = AssessmentFinalDecision::DISQUALIFIED->name;
+            }
+        }
+
+        // While unlikely, current step could go over.
+        // So, set it back to total steps
+        if ($currentStep > $totalSteps) {
+            $currentStep = $totalSteps;
+        }
+
+        return [
+            'currentStep' => $currentStep,
+            'finalDecision' => $finalDecision,
```

Comment:
Hmm. It occurs to me now that "finalDecision" and "stepDecisions" are kindof misnomers... Neither of these fields is ever decided directly by a user - they're computed based on other fields. "finalDecision" especially is going to be confused with the "Final decision" that users actually make when they screen a candidate in or out, which is stored in poolCandidateStatus.

Reply:
Yeah, I was kind of struggling with the naming here for the exact reason. Do you have any suggestions? I was thining of `assessmentDecision` but that exists on the `AssessmentResult` as well and could be conflated with that :disappointed: 

We just have so many layers to "decision" :sweat_smile: . Maybe the following works? Idk, I am at a loss :stuck_out_tongue: 

```
type AssessmentStatus {
   currentStep: Int
   qualificationStatus: AssessmentQualificationStatus! 
   stepAssessments: {
      step: UUID!
      decision: AssessmentDecision!
   }
}
```

Reply:
Yeah its a tough one. 😅 
`assessmentStepStatus` and `overallAssessmentStatus`?
`computedDecision` and `overallComputedDecision`?

Reply:
Thanks for the suggestions, updated in [refactor names for accuracy](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10642/commits/b776ef4857c025be6fee39c47a26e58773fd719a)